libdnf/base/transaction.cpp

```diff
@@ -87,10 +95,33 @@ Transaction::Impl::~Impl() {
     }
 }
 
+Transaction::Impl::Impl(Transaction & transaction, const BaseWeakPtr & base)
+    : transaction(&transaction),
+      base(base),
+      rpm_signature(base) {}
+
+Transaction::Impl::Impl(Transaction & transaction, const Impl & src)
+    : transaction(&transaction),
+      base(src.base),
+      libsolv_transaction(src.libsolv_transaction ? transaction_create_clone(src.libsolv_transaction) : nullptr),
```

Comment:
This particular block is only copied from `.hpp` here. I agree with you, also don't like overusing ternary operators for anything other than very simple expressions. But looking into `libsolv`, it seems that any such checking for `NULL` inputs in other `extern` functions are not present, so I doubt the maintainer would fix this specific case, but anyway I will give it a go :slightly_smiling_face:

Reply:
Hmm I was checking again through the `libsolv` codebase and it seems the failing hard and early for any `NULL` input parameter is a standard there, so I guess I'd leave it as it is. I don't think this can be considered as a bad or good way of handling this in `libsolv` and if it is treated in a consistent way, I would respect it.